### ISSUE TITLE
Get_data error with logit glmer model and weights

### ISSUE THREAD

**Issue by trifle** [easystats/insight#418]
Hi, I've encountered a regression in insight.

I'll try and add reproduction code in a second.
Somewhere between 0.11.1 and 0.14.3 as well as master (only tested those three so far), my glmer logit models start failing with 

```R
Error in `[.data.frame`(get_data(x, verbose = verbose), , rn, drop = FALSE) : 
  undefined columns selected
```

The model is approximately
```R
glmer(count/n ~ year * bool + (year | bool2), weights=n, data=data, family = binomial(link = "logit"))
```

The issue seems to lie in the scope where y is defined as a division, because the following mtcars example reproduces.
Note that I did not get the error without the weights, so I'd expect that parameter is involved.
```R
glmer(vs/cyl ~ disp  + (1 | cyl), data=mtcars, weights = gear, family = binomial(link = "logit"))
```

(This has singular fit, but I wasn't able to find a successful mixed logit model for mtcars)

[Edit: Reprex]

``` r
data("mtcars")
library(lme4)
#> Loading required package: Matrix
library(insight)
demo = glmer(vs/cyl ~ disp  + (1 | cyl),
             data=mtcars, weights = gear, family = binomial(link = "logit"))
#> Warning in eval(family$initialize, rho): non-integer #successes in a binomial
#> glm!
#> boundary (singular) fit: see ?isSingular
insight::model_info(demo)
#> Error in `[.data.frame`(get_data(x, verbose = verbose), , rn, drop = FALSE): undefined columns selected
```

<sup>Created on 2021-08-31 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by trifle** [comment]
PS: Perhaps this is a suitable "scary creature for the depth of my mind" for easystats/circus.

**Comment by strengejacke** [comment]
Should work now.

@mattansb @bwiernik related to https://github.com/easystats/performance/issues/165, what would be the expected number of observations for this kind of model?

``` r
data("mtcars")
library(lme4)
#> Loading required package: Matrix
library(insight)
demo <- glmer(vs/cyl ~ disp  + (1 | cyl),
              data=mtcars, weights = gear, family = binomial(link = "logit"))
#> Warning in eval(family$initialize, rho): non-integer #successes in a binomial
#> glm!
#> boundary (singular) fit: see ?isSingular

n_obs(demo)
#> [1] 118
nobs(demo)
#> [1] 32
```

<sup>Created on 2021-09-14 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by bwiernik** [comment]
tl;dr: I think we should add a `disaggregate` argument or similar to `n_obs()` to give our current behavior, but default to the result aggregate row count given by `nobs()`.

As this is aggregate data, `nobs()` is returning the number of unique covariate vectors (rows of data), `n_obs()` is returning the total number of trials (determined by summing the results of `weights()` model for aggregated data, which will be either the `weights` input for proportion success response as in this case or the row sums of the response matrix if matrix response). 

The reason we started the discussion where we changed the `n_obs()` behavior was to try to ensure various model performance statistics (AIC, various R2) performed equivalently. R omits constants from the log likelihood for a variety of models, which is why  the deviance differs depending on whether the model is estimated with aggregated data or not. 

At the time, I was thinking we might want re-compute the log-likelihood for aggregated models so that they match the value for individual (Bernoulli) models: `ll_individual = ll_aggregated + sum(lchoose(n_total, n_success))`, where n_total is the vector of trials and n_success is the vector of successes for each covariate pattern. `lchoose()` is the log-combination function. We would then be able to compute similarly matching values for AIC, BIC, R2, etc.

Thinking about it now, I don't know that this is really necessary for AIC, BIC--the differences between models with the same data structure will be the same, and doing just this adjustment still won't give analogous point-wise diagnostic statistics (deviance residuals, PSIS k values). So, there isn't much reason to deviate from the general R behavior. 

It's different for R2--there we probably do want to convert the binomial likelihoods to equivalent bernoulli likelihood so that we can accurately calculate `r2_nagelkerke()`, etc. For that, it would be useful to be able to quickly get the total trials count from `n_obs()`. 

Given the above considerations, I think we should go back to returning the results that match `nobs()` by default in `n_obs()`, but add a `disaggregate` argument or similar to return the total number of trials.
